clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchResponse.java

```diff
@@ -88,19 +88,35 @@ public OffsetFetchResponse(Map<TopicPartition, PartitionData> responseData) {
                 partitionData.set(PARTITION_KEY_NAME, partitionEntry.getKey());
                 partitionData.set(COMMIT_OFFSET_KEY_NAME, fetchPartitionData.offset);
                 partitionData.set(METADATA_KEY_NAME, fetchPartitionData.metadata);
-                partitionData.set(ERROR_CODE_KEY_NAME, fetchPartitionData.errorCode);
+                partitionData.set(PARTITION_ERROR_CODE_KEY_NAME, fetchPartitionData.errorCode);
                 partitionArray.add(partitionData);
             }
             topicData.set(PARTITIONS_KEY_NAME, partitionArray.toArray());
             topicArray.add(topicData);
         }
-        struct.set(RESPONSES_KEY_NAME, topicArray.toArray());
+
+        return topicArray;
+    }
+
+    /**
+     * Constructor for version 0 and 1
+     * @param responseData Fetched offset information grouped by topic-partition
+     */
+    public OffsetFetchResponse(Map<TopicPartition, PartitionData> responseData, int version) {
+        super(new Struct(ProtoUtils.responseSchema(ApiKeys.LIST_OFFSETS.id, version)));
+        //super(new Struct(CURRENT_SCHEMA));
+
+        struct.set(RESPONSES_KEY_NAME, getTopicArray(responseData).toArray());
         this.responseData = responseData;
+        // this constructor is used by api version 2 and above only if there is no top level errors,
+        // so it is safe to set the errorCode to None. The error code is not part of the response
+        // schema for versions 0 and 1, so setting it here has no impact for those versions.
+        this.errorCode = Errors.NONE.code();
```

Comment:
You're right. I'll try to follow a similar pattern for `OffsetFetchResponse` in the next update.